validate.go

```diff
@@ -208,6 +211,23 @@ func checkLinux(spec rspec.Spec) {
 			logrus.Fatalf("namespace %v is invalid.", spec.Linux.Namespaces[index])
 		} else if spec.Linux.Namespaces[index].Type == rspec.UTSNamespace {
 			utsExists = true
+		} else if spec.Linux.Namespaces[index].Type == rspec.IPCNamespace {
+			ipcExists = true
```

Comment:
On Mon, Jun 27, 2016 at 11:58:28PM -0700, Ma Shimiao wrote:
> Some items in config file need to work with some namespace.  If some
> namespaces did not exist, those items would not make sense.

The container process will always be in *some* namespace (for each
type).  Without a namespace entry, it will be in the runtime namespace
[1].  With a namespace entry that sets ‘path’, it will join some
existing namespace.  With a namespace entry that does not set ‘path’,
it will create a new namespace.  The current spec wording requires a
new namespace “if the config specifies anything else related to that
namespace”.  I think that means you need an IPC namespace without
‘path’ and a mount namespace without ‘path’ before you can set an
‘fs.mqueue.…’ sysctl.

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc1/glossary.md#runtime-namespace